http4k-testing-chaos/src/main/kotlin/org/http4k/chaos/ChaosBehaviours.kt

```diff
@@ -0,0 +1,69 @@
+package org.http4k.chaos
+
+import org.http4k.core.Request
+import org.http4k.core.Response
+import org.http4k.core.Status
+import java.lang.Thread.sleep
+import java.time.Duration
+import java.util.concurrent.ThreadLocalRandom
+
+interface ChaosBehaviour {
+    val description: String
+        get() = this.javaClass.name
+
+    fun inject(request: Request): Request = request
+    fun inject(response: Response): Response = response
+}
+
+class AdditionalLatencyBehaviour(
+    private val minDelay: Duration,
+    private val maxDelay: Duration
+) : ChaosBehaviour {
+    override val description = "Latency"
+
+    override fun inject(response: Response): Response {
+        val delay = ThreadLocalRandom.current().nextInt(minDelay.toMillis().toInt(), maxDelay.toMillis().toInt())
+        sleep(delay.toLong())
+        return response
+    }
+
+    companion object {
+        fun fromEnv(): AdditionalLatencyBehaviour {
+            val minDelay = Duration.parse(System.getenv("CHAOS_LATENCY_MS_MIN") ?: "PT0.1S")
+            val maxDelay = Duration.parse(System.getenv("CHAOS_LATENCY_MS_MAX") ?: "PT0.5S")
+            return AdditionalLatencyBehaviour(minDelay, maxDelay)
+        }
+    }
+}
+
+class ChaosException(message: String) : Exception(message)
+
```

Comment:
generally, the pattern within the project is for this kind of thing to appear as an extension function of the parent type. So, for instance `fun ChaosBehaviour.ThrowException()`. This one in particular could take an exception as an arg, and default to throw ChaosException as the default.